aws/src/test/java/org/apache/iceberg/aws/glue/TestGlueCatalog.java

```diff
@@ -585,4 +589,21 @@ public void testValidateIdentifierSkipNameValidation() {
         CATALOG_NAME, WAREHOUSE_PATH, props, glue, LockManagers.defaultLockManager(), null);
     Assert.assertEquals(glueCatalog.isValidIdentifier(TableIdentifier.parse("db-1.a-1")), true);
   }
+
+  @Test
+  public void testS3FileIOSerialization() throws IOException {
```

Comment:
This should not be a Glue test. It should be a `S3FileIO` test.

Reply:
Initially added this as a glue test considering only flow of S3FileIO initialization was problematic and newTableOps were protected ... 

adding handling of the immutable map in S3FileIO itself, definitely makes sense.

removed this test and added one for S3FileIO, thanks !